### PR TITLE
Add new redirect_uri for the new Check-in app

### LINKED ISSUE #6039: Update the system Check-in App with a new redirect_uri
The Check-in `OAuthApplication` in Indico needs to be updated with a new redirect-uri for the new PWA app once we decide on what domain to use for it. I'm just putting this here so we don't forget to do it before releasing the new app.

Todo:
- Remove `redirect_uris` from `__enforced_data__`
- Add the new redirect to `__default_data__` (this only affects new installations)
- Add an alembic revision to change the redirect uri on existing installations

### PR DESCRIPTION
**Pull request by tomasr8**
Closes #6039

### CODE REVIEW

File: indico/migrations/versions/20231124_0941_fb0ca1440185_add_redirect_uri_checkin_app.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+down_revision = '0acf26d68434'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute(f'''
+        UPDATE oauth.applications
+        SET redirect_uris = array_append(redirect_uris, 'https://checkin.getindico.io')
+        WHERE system_app_type = {SystemAppType.checkin.value}''')  # noqa: S608
```
Comment (ThiefMaster) [inline indico/migrations/versions/20231124_0941_fb0ca1440185_add_redirect_uri_checkin_app.py:29]: wrap before the `'''`

PS: I'd just hardcode this enum value...

**Comment by ThiefMaster** [comment]
Maybe you could also add the `CHECKIN_APP_URL` indico.conf setting (default to `https://checkin.getindico.io`) in the same PR, and show it somewhere in the event area?

**Comment by tomasr8** [comment]
How about something like this?

image

**Comment by ThiefMaster** [comment]
hmm why not next to the button to show the setup QR code, or in the thingy that opens where the QR code is?

**Comment by tomasr8** [comment]
Better like this? The link is only visible when tickets are enabled (same as the qr code button)
image

**Comment by ThiefMaster** [comment]
hmm why a link and not a button? feels a bit inconsistent

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**